abci/abci.go

```diff
@@ -58,21 +58,33 @@ func NewProposalHandler(
 func (h *ProposalHandler) PrepareProposalHandler() sdk.PrepareProposalHandler {
 	return func(ctx sdk.Context, req abci.RequestPrepareProposal) abci.ResponsePrepareProposal {
 		var (
-			selectedTxs  [][]byte
-			totalTxBytes int64
+			selectedTxs   [][]byte
+			totalTxBytes  int64
+			totalGasLimit int64
 		)
 
 		bidTxIterator := h.mempool.AuctionBidSelect(ctx)
 		txsToRemove := make(map[sdk.Tx]struct{}, 0)
 		seenTxs := make(map[string]struct{}, 0)
 
+		// Do we need to deduct some from here since begin block gas gets carried over?
```

Comment:
Upstream bug apparent on most `v0.47.x` chains makes me think we want to deduct some amount from here.

Reply:
It gets carried over into DeliverTxState's gas meter, I don't think that the PrepareProposal gas meter shld have anything to do here